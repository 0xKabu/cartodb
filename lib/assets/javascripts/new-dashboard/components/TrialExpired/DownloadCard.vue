<template>
  <div class="card-row">
    <div class="card-cell cell--start">
      <div class="row-image" v-if="dataType">
          <div class="icon" :class="`icon--${dataType}`"></div>
      </div>
      <div class="row-image row-image--map row-image--large" v-if="image">
          <div class="icon icon--image" :style="`background-image: url('${image}')`"  @error="onThumbnailError" v-if="!isThumbnailErrored"></div>
      </div>
    </div>
    <div class="card-cell cell--main">
      <div class="title-container">
        <h3 class="text is-caption is-txtGrey u-ellipsis row-title" :title='name'>
          {{name}}
        </h3>
      </div>
    </div>
    <div class="card-cell cell--download">
      <a class="title is-small is-txtPrimary" href="#">Download</a>
    </div>
  </div>
</template>

<script>
export default {
  name: 'DownloadCard',
  props: {
    name: String,
    dataType: String,
    image: String
  },
  methods: {
    onThumbnailError () {
      this.isThumbnailErrored = true;
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.card-row {
  display: flex;
  align-items: center;
  width: 100%;
  height: 80px;
  padding: 0 24px 0 14px;
  background-color: $white;
}

.row-image {
  width: 36px;
  height: 36px;
  padding: 9px;
  overflow: hidden;
  transition: all 0.25s cubic-bezier(0.4, 0.01, 0.165, 0.99);
  border-radius: 4px;
  background-color: $light-grey;

  .row-typeIcon {
    fill: $text-color;
  }

  &.row-dataType--premium {
    background-color: $premium-color;

    .row-typeIcon {
      fill: $white;
    }
  }

  &.row-image--large {
    width: 48px;
    height: 48px;
    padding: 0;
  }

  &.row-image--map {
    background: url($assetsDir + '/images/layout/default-map-bkg.png') no-repeat center 0;
  }
}

.card-cell {
  position: relative;
  flex-grow: 0;
  flex-shrink: 0;
  padding: 0 10px;
  text-overflow: ellipsis;
  white-space: nowrap;

  &:first-of-type {
    padding-left: 0;
  }

  &:last-of-type {
    padding-right: 0;
  }
}

.cell--start {
  display: flex;
  align-items: center;
  align-self: flex-start;
  height: 100%;
  overflow: hidden;
}

.cell--main {
  flex-grow: 1;
  flex-shrink: 1;
  min-width: 200px;
}

.cell--download {
  display: flex;
  align-items: center;
}

.icon {
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-position: center;

  &.icon--point {
    background-image: url("../../assets/icons/datasets/data-types/dots.svg");
  }

  &.icon--polygon {
    background-image: url("../../assets/icons/datasets/data-types/area.svg");
  }

  &.icon--line {
    background-image: url("../../assets/icons/datasets/data-types/line.svg");
  }

  &.icon--empty,
  &.icon--unknown {
    background-image: url("../../assets/icons/datasets/data-types/unknown.svg");
  }

  &.icon--image {
    background-size: cover;
  }
}

.title-container {
  display: flex;
  align-items: center;
}
</style>
