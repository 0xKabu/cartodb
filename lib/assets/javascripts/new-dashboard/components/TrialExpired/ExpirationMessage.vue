<template>
  <section class="section section--noBorder">
    <div class="container warning-container">
      <div class="warning-icon">
        <img svg-inline src="../../assets/icons/expired-trial/warning-icon.svg">
      </div>
      <h2 class="warning-title title is-title is-txtDarkBlue">Your trial has expired</h2>
      <p class="warning-description text is-body is-txtDarkBlue">We couldnâ€™t log you in because your trial expired on September 10, 2018.<br/> All of your data is safe, but no one will be able to log in until you account is upgraded.</p>
      <div class="buttons-container">
        <a href="#" class="button is-bgDarkBlue button--large upgrade-button">
          <span>Upgrade your account</span>
        </a>
        <a href="#" class="buttonLink no-transform button--large is-txtDarkBlue">
          <span>What happens now?</span>
        </a>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'ExpirationMessage',
  components: {
  },
  computed: {
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.upgrade-button {
  margin-right: 64px;
}

.warning-container {
  text-align: center;
}

.warning-icon {
  margin-bottom: 48px;
}

.warning-title {
  margin-bottom: 16px;
}

.buttons-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 36px;
}
</style>
