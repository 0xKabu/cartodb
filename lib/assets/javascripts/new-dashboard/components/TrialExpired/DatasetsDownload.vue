<template>
<section class="section section--noBorder">
  <div class="container grid">
    <div class="full-width">
      <SectionTitle class="grid-cell" title="Download your Data">
        <template slot="icon">
          <img src="../../assets/icons/section-title/map.svg">
        </template>
      </SectionTitle>
      <ul class="grid">
        <li class="grid-cell grid-cell--col12 download-element" v-for="dataset in datasets" :key="dataset.id">
          <DownloadCard :name="dataset.name" :dataType="dataType(dataset)"></DownloadCard>
        </li>
      </ul>
    </div>
  </div>
</section>
</template>

<script>
import SectionTitle from 'new-dashboard/components/SectionTitle';
import DownloadCard from 'new-dashboard/components/TrialExpired/DownloadCard';
import { mapState } from 'vuex';

export default {
  name: 'DatasetsDownload',
  components: {
    SectionTitle,
    DownloadCard
  },
  computed: {
    ...mapState({
      datasets: state => state.datasets.list
    })
  },
  methods: {
    dataType (dataset) {
      const geometryTypes = {
        'st_multipolygon': 'polygon',
        'st_polygon': 'polygon',
        'st_multilinestring': 'line',
        'st_linestring': 'line',
        'st_multipoint': 'point',
        'st_point': 'point',
        '': 'empty'
      };
      let geometry = '';
      if (dataset.table && dataset.table.geometry_types && dataset.table.geometry_types.length) {
        geometry = dataset.table.geometry_types[0];
      }
      const currentGeometryType = geometry.toLowerCase();
      return geometryTypes[currentGeometryType] ? geometryTypes[currentGeometryType] : 'unknown';
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.full-width {
  width: 100%;
}

.download-element {
  border-bottom: 1px solid #F2F6F9;
}
</style>
