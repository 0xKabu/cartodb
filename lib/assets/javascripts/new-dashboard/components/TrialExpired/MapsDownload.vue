<template>
<section class="section section--noBorder">
  <div class="container grid">
    <div class="full-width">
      <SectionTitle class="grid-cell" title="Download your Maps">
        <template slot="icon">
          <img src="../../assets/icons/section-title/map.svg">
        </template>
      </SectionTitle>
      <ul class="grid">
        <li class="grid-cell grid-cell--col12 download-element" v-for="map in maps" :key="map.id">
          <DownloadCard :name="map.name" :image="mapThumbnailUrl"></DownloadCard>
        </li>
      </ul>
    </div>
  </div>
</section>
</template>

<script>
import { mapState } from 'vuex';
import SectionTitle from 'new-dashboard/components/SectionTitle';
import DownloadCard from 'new-dashboard/components/TrialExpired/DownloadCard';
import * as Visualization from 'new-dashboard/core/visualization';

export default {
  name: 'MapsDownload',
  components: {
    SectionTitle,
    DownloadCard
  },
  computed: {
    ...mapState({
      maps: state => state.maps.list
    }),
    mapThumbnailUrl () {
      return Visualization.getThumbnailUrl(this.$props.map, this.$cartoModels, { width: 600, height: 280 });
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.full-width {
  width: 100%;
}

.download-element {
  border-bottom: 1px solid #F2F6F9;
}
</style>
