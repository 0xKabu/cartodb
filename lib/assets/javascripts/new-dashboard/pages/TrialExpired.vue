<template>
  <section class="page">
    <ExpirationMessage :expirationDays="expirationDays" :upgradeURL="upgradeURL"></ExpirationMessage>
  </section>
</template>

<script>
import ExpirationMessage from 'new-dashboard/components/TrialExpired/ExpirationMessage';

export default {
  name: 'TrialExpired',
  components: {
    ExpirationMessage
  },
  data () {
    return {
      expirationDays: window.expiration_days,
      upgradeURL: window.upgrade_url
    };
  }
};
</script>
