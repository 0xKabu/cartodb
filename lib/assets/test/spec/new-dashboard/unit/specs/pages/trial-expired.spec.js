import { shallowMount } from '@vue/test-utils';
import TrialExpired from 'new-dashboard/pages/TrialExpired';

const $t = key => key;

describe('TrialExpired.vue', () => {
  it('should render correct contents', () => {
    const data = function () {
      return {
        expirationDays: 30,
        upgradeURL: 'https://user.carto.com/plan'
      };
    };

    const page = shallowMount(TrialExpired, {
      data,
      mocks: {
        $t
      }
    });

    expect(page).toMatchSnapshot();
  });
});
