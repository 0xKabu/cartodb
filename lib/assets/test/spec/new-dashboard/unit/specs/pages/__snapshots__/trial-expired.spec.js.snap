// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialExpired.vue should render correct contents 1`] = `
<section class="page">
  <expirationmessage-stub expirationdays="30" upgradeurl="https://user.carto.com/plan"></expirationmessage-stub>
</section>
`;
